components/fms-server/src/main.rs

```diff
@@ -130,7 +129,81 @@ async fn get_vehicles(
             Json(result_object)
         })
         .map_err(|e| {
-            error!("error retrieving vehicle status: {e}");
+            error!("error retrieving vehicles: {e}");
             StatusCode::INTERNAL_SERVER_ERROR
         })
 }
+
+fn parse_latest_only(params: &HashMap<String, String>) -> Result<Option<bool>, StatusCode> {
+    let latest_parameter = params.get("latestOnly");
+    if let Some(latest_string) = latest_parameter {
+        let latest_result = latest_string.parse();
+        if latest_result.is_err() {
+            return Err(StatusCode::BAD_REQUEST);
+        }
+        return Ok(latest_result.ok());
+    }
+    Ok(None)
+}
+
+async fn get_vehiclesstatuses(
+    State(influx_server): State<Arc<InfluxReader>>,
+    Query(params): Query<HashMap<String, String>>,
+) -> Result<Json<serde_json::Value>, StatusCode> {
+    
+    
+    let start_parameter = params.get("starttime");
+    let stop_parameter = params.get("stoptime");
+    
+    
+    let latest_only = match parse_latest_only(&params) {
+        Ok(value) => value,
+        Err(status) => return Err(status), 
+    };
+    
+    if start_parameter.is_none() && latest_only.is_none() {
+        // rFMS makes it mandatory to either supply the starttime or latestOnly
+        return Err(StatusCode::BAD_REQUEST);
+    }
+
+    if latest_only.is_some() && (start_parameter.is_some() || stop_parameter.is_some()) {
+        // rFMS does not allow to set latestOnly and and time at the same time
+        return  Err(StatusCode::BAD_REQUEST);
+    }
+
+    let start_time = start_parameter.map_or(0, |text| {
+        DateTime::<Utc>::from_str(text).map_or(0, |time| time.timestamp())
+    });
+
+    let stop_time = stop_parameter
+        .map_or_else(|| Utc::now().timestamp(), |text| {
+            DateTime::<Utc>::from_str(text).map_or_else(|_| Utc::now().timestamp(), |time| time.timestamp())
+        });
+
+    let vin = params.get("vin");
+    let trigger_filter = params.get("triggerFilter");
```

Comment:
Is this query parameter checking/parsing something that is needed for the other endpoints as well? If so, can we reuse/share this code?

Reply:
Yes, that it is something that can be shared. My idea was to add this sharing in a later PR and limit the scope of this PR to the initial addition of the /vehiclestatuses endpoint. 
Should I add the sharing to this PR instead?